src/function/scalar/string/substring.cpp

```diff
@@ -4,72 +4,15 @@
 #include "duckdb/common/vector_operations/vector_operations.hpp"
 #include "duckdb/common/vector_operations/ternary_executor.hpp"
 #include "utf8proc.hpp"
+#include "util.hpp"
+#include <algorithm>
+
+#include <iostream>
 
 using namespace std;
 
 namespace duckdb {
 
-static string_t substring_ascii_only(Vector &result, const char *input_data, int offset, int length) {
-	auto result_string = StringVector::EmptyString(result, length);
-	auto result_data = result_string.GetData();
-	memcpy(result_data, input_data + offset, length);
-	result_string.Finalize();
-	return result_string;
-}
-
-static string_t substring_scalar_function(Vector &result, string_t input, int offset, int length,
```

Comment:
Could you move the definitions back here, and expose this method as a a static method in `
struct SubstringFun` (`static string_t ScalarFunction(Vector& result, string_t input, ...)`) rather than adding the `util` header?

It can then be used in the LEFT/RIGHT functions by calling `SubstringFun::ScalarFunction(result, input, ...)`

Reply:
Done, PTAL